### DISCUSSION TITLE
ConfigureHttpClient not firing

### DISCUSSION THREAD

**Discussion by alin-devnet** [OrchardCMS/OrchardCore#16314] [Q&A]
### Orchard Core version

1.8.x

### To Reproduce

When adding a http client binding, the ConfigureHttpClient is not firing.

`services.AddHttpClient<IService, Service>()
    .**ConfigureHttpClient**(c => c.BaseAddress = new Uri("test-uri"));`

This seems to work fine if the binding is added in ConfigureServices.

` builder.Services
     .AddOrchardCore()
     .WithTenants()
     .ConfigureServices((m, serviceProvider) =>
     {
        services.AddHttpClient<IService, Service>()
            .**ConfigureHttpClient**(c => c.BaseAddress = new Uri("test-uri"));
     });`

Whilst the above works, my understanding is that each tenant will have it's own HttpClient instance injected and considering each tenant will call the same API, it might end up not using resources optimally (thinking about snat ports here).

**Comment by Piedone** [comment]
This isn't really a bug. Due to the multi-tenant nature of OC yes, services by default are specific to tenants.

However, you can register services for the whole app too, but with `AddOrchardCms(builder => builder.ApplicationServices...)`.
